contracts/GovNftSugar.vy

```diff
@@ -57,116 +61,125 @@ interface IToken:
+def minted(_account: address, _collection: address) -> DynArray[GovNft, MAX_RESULTS]:
   """
-  @notice Returns all minted GovNFTs for the given account
+  @notice Returns all minted GovNFTs for the given account and collection
   @return Array of GovNft structs
   """
-  return self._minted(_account)
+  return self._minted(_account, _collection)
 
 @internal
 @view
-def _minted(_account: address) -> DynArray[GovNft, MAX_RESULTS]:
+def _minted(_account: address, _collection: address) -> DynArray[GovNft, MAX_RESULTS]:
   """
-  @notice Returns all minted GovNFTs for the given account
+  @notice Returns all minted GovNFTs for the given account and collection
   @return Array of GovNft structs
   """
   govnfts: DynArray[GovNft, MAX_RESULTS] = empty(DynArray[GovNft, MAX_RESULTS])
 
-  supply: uint256 = self.nft.totalSupply()
+  nft: IGovNFT = IGovNFT(_collection)
+  supply: uint256 = nft.totalSupply()
 
   for govnft_id in range(1, MAX_NFTS):
     if govnft_id > supply:
       break
-    lock: Lock = self.nft.locks(govnft_id)
+    lock: Lock = nft.locks(govnft_id)
 
     if lock.minter == _account:
-      govnft: GovNft = self._byId(govnft_id)
+      govnft: GovNft = self._byId(govnft_id, _collection)
       govnfts.append(govnft)
 
   return govnfts
 
 @external
 @view
-def byId(_govnft_id: uint256) -> GovNft:
+def byId(_govnft_id: uint256, _collection: address) -> GovNft:
   """
-  @notice Returns GovNFT data based on ID
-  @param _govnft_id The GovNFT ID to look up
+  @notice Returns GovNFT data based on ID and collection
+  @param _govnft_id The GovNFT ID and collection to look up
   @return GovNft struct
   """
-  return self._byId(_govnft_id)
+  return self._byId(_govnft_id, _collection)
 
 @internal
 @view
-def _byId(_govnft_id: uint256) -> GovNft:
+def _byId(_govnft_id: uint256, _collection: address) -> GovNft:
   """
-  @notice Returns GovNFT data based on ID
-  @param _govnft_id The GovNFT ID to look up
+  @notice Returns GovNFT data based on ID and collection
+  @param _govnft_id The GovNFT ID and collection to look up
   @return GovNft struct
   """
-
-  lock: Lock = self.nft.locks(_govnft_id)
+  nft: IGovNFT = IGovNFT(_collection)
+  lock: Lock = nft.locks(_govnft_id)
 
   token_addr: address = lock.token
   token: IToken = IToken(token_addr)
 
   return GovNft({
     id: _govnft_id,
     total_locked: lock.total_locked,
-    amount: self.nft.locked(_govnft_id),
+    amount: nft.locked(_govnft_id),
     total_claimed: lock.total_claimed,
-    claimable: self.nft.unclaimed(_govnft_id),
+    claimable: nft.unclaimed(_govnft_id),
     split_count: lock.split_count,
     cliff_length: lock.cliff_length,
     start: lock.start,
     end: lock.end,
     token: token_addr,
     vault: lock.vault,
     minter: lock.minter,
-    owner: self.nft.ownerOf(_govnft_id),
-    address: self.nft.address,
+    owner: nft.ownerOf(_govnft_id),
+    address: _collection,
     delegated: token.delegates(lock.vault)
```

Comment:
I might be a good idea to wrap the delegates call around a try/catch or something similar in case the lock token is not a governance token. The call will fail if it isn't.